### PR TITLE
chore: bump dependencies to latest version

### LINKED ISSUE #351: OpenAPI specifications pages have no sidebar
The most updated version of the plugin has an issue where the sidebar is unable to render (see https://github.com/withastro/starlight/discussions/982). We need to track this issue's progress on the Starlight side and update once they've resolved. *Update*: Plugin support merged into 0.14.0 (see https://github.com/withastro/starlight/pull/942)

On the Starlight OpenAPI integration side: https://github.com/HiDeoo/starlight-openapi/issues/11
Continued exploration: https://github.com/withastro/starlight/pull/1175

### LINKED ISSUE #388: Unable to upgrade to Astro v4 / Starlight 0.15.0 due to plugin issue
See https://github.com/HiDeoo/starlight-openapi/issues/14
Will continue to monitor

### PR DESCRIPTION
**Pull request by huijing**
## Changes proposed in this pull request

- This PR updates dependencies to the latest versions, which finally resolves the sidebar problem

## Context

Closes #351 
Closes #388

### CODE REVIEW

**Review by melissahenderson — APPROVED** [review]

File: docs/astro.config.mjs [hunk 1]
```diff
@@ -29,8 +10,11 @@ export default defineConfig({
       title: 'Open Payments',
       description:
         'An API for open access to financial accounts to send and receive payments.',
+      components: {
+        Header: './src/components/Header.astro'
+      },
       customCss: [
-        './node_modules/@interledger/docs-design-system/src/styles/green-theme.css',
+        './node_modules/@interledger/docs-design-system/src/styles/teal-theme.css',
```
Comment (mkurapov) [inline docs/astro.config.mjs:17]: I see we are getting the teal makeover here 💅

Comment (mkurapov) [inline docs/astro.config.mjs:17]: The only thing I would flag here is, is switching to the teal theme responsible for causing the darker grey code blocks in the API references?

Screenshot 2024-02-19 at 15 33 38

**Review by mkurapov — APPROVED** [review]
The only thing worth noting is

File: docs/astro.config.mjs (hunk 1 above)
Comment (huijing) [reply docs/astro.config.mjs:17]: no, as part of the plugin update, it switched over to expressive code, and there are some style settings that will need to be over-ridden. I might escalate those into the shared design system, especially if Rafiki also wants to display OpenAPI docs.

**Pull request merged**